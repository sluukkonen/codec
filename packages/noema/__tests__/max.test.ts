import { max, number } from "../src/index.js"
import { expectParseFailure, expectParseSuccess } from "./helpers.js"

test("should parse if value is less than or equal to the minimum", () => {
  const nonPositiveNumber = max(number, 0)
  expectParseSuccess(nonPositiveNumber, 0)
  expectParseSuccess(nonPositiveNumber, -1)
})

test("should fail to parse if value is greater than the minimum", () => {
  const nonPositiveNumber = max(number, 0)
  expectParseFailure(nonPositiveNumber, 1)
})

test("should fail to parse if value isn't parsed by the underlying codec", () => {
  const nonPositiveNumber = max(number, 0)
  expectParseFailure(nonPositiveNumber, "")
})
