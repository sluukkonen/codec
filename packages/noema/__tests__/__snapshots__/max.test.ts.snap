// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`min should fail to parse if value is greater than the minimum 1`] = `
[
  {
    "code": "too_large",
    "max": 0,
    "path": [],
  },
]
`;

exports[`min should fail to parse if value isn't parsed by the underlying codec 1`] = `
[
  {
    "code": "invalid_type",
    "expected": "number",
    "path": [],
    "received": "string",
  },
]
`;
